Lib/enum.py

```diff
@@ -965,38 +1161,39 @@ def __contains__(self, other):
             raise TypeError(
                 "unsupported operand type(s) for 'in': '%s' and '%s'" % (
                     type(other).__qualname__, self.__class__.__qualname__))
+        if other._value_ == 0 or self._value_ == 0:
+            return False
         return other._value_ & self._value_ == other._value_
 
     def __iter__(self):
         """
         Returns flags in decreasing value order.
         """
-        members, extra_flags = _decompose(self.__class__, self.value)
-        return (m for m in members if m._value_ != 0)
+        return (
+                m
+                for m in sorted(
+                    self.__class__, key=lambda m: m._value_, reverse=True
+                    )
```

Comment:
1. The order of the class should be definition order (`list(Flag)`), but I find that a `repr()` in big-endian format is more useful (`list(Flag.member)`).
2. Would this support a member with unknown bits?
3. A generator option sounds interesting, and might be fine for `__iter__`, but I would still want `repr()` to be as-is (which should be fine -- if one is doing a `repr()` then speed shouldn't matter).

Reply:
2. Done.

Reply:
I don't agree about `repr()` or `str()` speed not being important.  Production systems rely on these for metrics reporting, logging, visualization, etc.

Reply:
That's a good point.  However, while speed is important, it's not the most important, or we wouldn't have `Enum` to begin with.

I also see that iterating from msb to lsb is leaking an implementation detail, but so is lsb to msb.  Really, it should be iterating in member definition order, just like Enum does.

So what I'll do is:  if the `Flag` 's definition order is lsb to msb, then the more efficient iterator you proposed is used, otherwise a less efficient iterator that still honors definition order is used.

For the definition-order iterator I can either check every flag against the target, or use the lsb iterator and sort afterwards using an added `_def_order_` attribute.

Reply:
Summarizing changes:

1. Iteration order is definition order.
2. Iteration is over bits in the member instead of bits in the class.
3. Member iteration is also in definition order, but if definition order is increasing value order then the more efficient lsb algorithm is used.

Reply:
>  Really, it should be iterating in member definition order, just like Enum does.

I kind of followed why you wanted Flag class iter to be in definition order, following Enum.

But I don't understand why it's important to follow definition order when iterating the individual elements of a composite value.  Flag is a set (in my view), and a set doesn't have ordering-- so iterate in whatever order is efficient for the implementation.

> So what I'll do is: if the Flag 's definition order is lsb to msb, then the more efficient iterator you proposed is used, otherwise a less efficient iterator that still honors definition order is used.

It's unfortunately more complication for both the implementation and the user.

Reply:
I see it as less complication for both: the "correct" iterator is chosen at class definition, then never changes; user's always get definition order.  How is that more complication?